Classes/KIFUITestActor.m

```diff
@@ -622,22 +622,38 @@ - (void)selectPickerValue:(NSArray *)pickerColumnValues pickerType:(KIFPickerTyp
         }
         // Find the picker view
         UIPickerView *pickerView = nil;
+        NSInteger firstRowToSearchIndex = 0;
         switch (pickerType)
         {
             case KIFUIDatePicker:
+            {
                 pickerView = [[[[UIApplication sharedApplication] datePickerWindow] subviewsWithClassNameOrSuperClassNamePrefix:@"UIPickerView"] lastObject];
                 KIFTestCondition(pickerView, error, @"No picker view is present");
+                
+                // If minimum date limit exist, search the value from minimum and beyond.
+                UIDatePicker *datePicker = (UIDatePicker *)pickerView;
+                if (datePicker.datePickerMode == UIDatePickerModeDateAndTime
+                    && datePicker.minimumDate != nil) {
+                    
+                    [datePicker setDate:datePicker.minimumDate];
+                    firstRowToSearchIndex = [pickerView selectedRowInComponent:0];
+                }
+                
+                
                 break;
+            }
             case KIFUIPickerView:
+            {
                 pickerView = [[[[UIApplication sharedApplication] pickerViewWindow] subviewsWithClassNameOrSuperClassNamePrefix:@"UIPickerView"] lastObject];
+            }
         }
         
         NSInteger componentCount = [pickerView.dataSource numberOfComponentsInPickerView:pickerView];
         KIFTestCondition(componentCount == columnCount, error, @"The UIDatePicker does not have the expected column count.");
         
         for (NSInteger componentIndex = 0; componentIndex < componentCount; componentIndex++) {
             NSInteger rowCount = [pickerView.dataSource pickerView:pickerView numberOfRowsInComponent:componentIndex];
-            for (NSInteger rowIndex = 0; rowIndex < rowCount; rowIndex++) {
+            for (NSInteger rowIndex = firstRowToSearchIndex; rowIndex < rowCount; rowIndex++) {
```

Comment:
Ah, nevermind, there are 10k rows for each component, not just for the first one. That is pretty bizarrely implemented by Apple. I guess this works as is.